ctlog/ctlog.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ctlog contains data structures and methods to do with CT Log metadata
+// that is needed by the monitor.
+//
+// TODO(katjoyce): Try to come up with a better package name.
+package ctlog
+
+import (
+	"crypto"
+	"fmt"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/logid"
+)
+
+// Log represents a CT Log and contains the Log metadata needed by the monitor.
+type Log struct {
+	ID        logid.LogID
+	Name      string
+	URL       string
+	PublicKey crypto.PublicKey
+}
+
+// New creates a Log structure, populating the fields appropriately.
+//
+// TODO(katjoyce): replace this implementation with something less hacky that
```

Comment:
Should be github user name here I think.

Reply:
OK it's not worth that much effort but I've tried to keep external work separate.